torch/fx/experimental/validator.py

```diff
@@ -510,9 +540,11 @@ def to_z3_boolean_expr(self, e: sympy.Basic) -> z3.BoolRef:
             ), f"expected boolean expression. Got: {z3expr}"
             return z3expr
 
-        def add_source_expr(self, e: z3.BoolRef) -> None:
+        def add_source_expr(self, e: Union[z3.BoolRef, z3.ArithRef]) -> None:
```

Comment:
Why is this needed?

Reply:
Boolean and integer bitwise ops go through the same implementation, which now usually returns an integer (except for that weird case with the failing benchmark that I'm still aiming to repro). But this part of the code expects a boolean expression, so I changed this function to accept integer/arith expression, with intent to convert into a boolean expression.